src/sentry/static/sentry/app/views/inviteMember/inviteMember.jsx

```diff
@@ -0,0 +1,228 @@
+                  key={id}
+                  onClick={() => allowed && this.setState({selectedRole: id})}
+                  style={allowed ? {} : {color: 'grey', cursor: 'default'}}>
+                  <label style={allowed ? {} : {cursor: 'default'}}>
+                    <Radio id={id} value={name} checked={id === selectedRole} readOnly />
+                    {name}
+                    <div className="help-block">{desc}</div>
+                  </label>
+                </li>
+              );
+            })}
+          </ul>
+        </div>
+      </div>
+    );
+  },
+
+  renderTeamSelect() {
+    let org = this.getOrganization();
+    let {teams} = org;
+    let {selectedTeams} = this.state;
+    //no need to select a team when there's only one option
+    if (teams.length < 2) return null;
+    return (
+      <div className="new-invite-team box">
+        <div className="box-header">
+          <h4>{t('Team') + ':'}</h4>
+        </div>
+        <div className="grouping-controls team-choices row box-content with-padding">
+          {teams.map(({slug, name, id}, i) => (
+            <div
+              key={id}
+              onClick={e => {
+                e.preventDefault();
+                this.toggleID(id);
+              }}
+              className="col-md-3">
+              <label className="checkbox">
+                <Checkbox id={id} value={name} checked={selectedTeams.has(id)} />
+                {name}
+                <span className="team-slug">{slug}</span>
+              </label>
+            </div>
+          ))}
+        </div>
+      </div>
+    );
+  },
+
+  render() {
+    let {error, loading, isInvite} = this.state;
+    return (
+      <div>
+        <h3>{t('Add Member to Organization')}</h3>
+        <p>
+          {isInvite
+            ? t(
+                'Invite a member to join this organization via their email address. If they do not already have an account, they will first be asked to create one. Multiple emails delimited by commas.'
+              )
+            : t(
+                'You may add a user by their username if they already have an account. Multiple inputs delimited by commas.'
+              )}
+        </p>
+        <div className={classnames({'has-error': error && error.email})}>
+          {loading && <LoadingIndicator mini className="pull-right" />}
+          <TextField
+            name="email"
+            label={isInvite ? t('Email') + '(s)' : t('Username') + '(s)'}
+            placeholderlabel="e.g. teammate@example.com"
+            spellCheck="false"
+            onChange={v => this.setState({email: v})}
+          />
+          {error && error.email && <p className="error">{error.email}</p>}
+        </div>
+        {error && error.role && <p className="error alert-error">{error.role}</p>}
+        {this.renderRoleSelect()}
+        {this.renderTeamSelect()}
+        <Button priority="primary" className="submit-new-team" onClick={this.submit}>
```

Comment:
I wonder if it'd be better to namespace class names that are only used as selectors for our tests. I could see someone thinking this is a class name with no associated css with it.

Reply:
do you mean naming this something like invite-member-submit?

Reply:
Ah no, sorry, was thinking more of a global prefix for selectors only used for tests. `test-`, or `js-`, idk.